### ISSUE TITLE
Repeated title in pretrain_lm (in read_wiki_articles)

### ISSUE THREAD

**Issue by abedkhooli** [n-waves/ulmfit-multilingual#26]
When running the ulmfit.train_lm and when wiki articles are read (separated by = title =, the resulting dataframe has the title repeated as 
= title = 
and 
title
Do we need both?

**Comment by PiotrCzapla** [comment]
nope, that is a bug, although it shouldn't matter much for downstream tasks. But can you fix it and write test for it?

**Comment by PiotrCzapla** [comment]
btw. @abedkhooli  I've just pushed a fix that you might be interested in if you are training lm on local wikipedia

**Comment by abedkhooli** [comment]
Thanks @PiotrCzapla I'll try the fix as I couldn't process large files and ended up splitting train tokens into 20 pieces (memory issues). For duplicate title, I used this (probably left an extra line behind: \n): 
```
    for i,line in enumerate(lines):
        if i > 0 and istitle(lines[i-1]):
            continue # skip second title 
        current_article += line
        if i < len(lines)-2 and lines[i+1] == ' \n' and istitle(lines[i+2]):
            articles.append(current_article)
            current_article = ''
    articles.append(current_article)
    print(f"Wiki text was split to {len(articles)} articles")
```

**Comment by PiotrCzapla** [comment]
@abedkhooli the code above does not split by article and it is what most likely caused the memory issue you had. The issue is fixed in master here: 0085c18ae057d0ca65ca40f127facab689f7ac2b

**Comment by PiotrCzapla** [comment]
We will update the way we download and store wikitext, so this code won't be relevant.
The work on is being done here:https://github.com/n-waves/ulmfit-multilingual/tree/multifit
